core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -624,6 +624,15 @@ public IRubyObject rewind() {
         return this;
     }
 
+    @JRubyMethod(name = "empty?", meta = true)
+    public static IRubyObject empty_p(ThreadContext context, IRubyObject recv, IRubyObject arg) {
+        Ruby runtime = context.runtime;
+        RubyString path = StringSupport.checkEmbeddedNulls(runtime, RubyFile.get_path(context, arg));
+        RubyFileStat fileStat = runtime.newFileStat(path.asJavaString(), false);
+        boolean isDirectory = fileStat.directory_p().isTrue();
```

Comment:
@headius @enebo 

I didn't have any changes planned. The code does throw a `Errno::ENOENT` in the correct scenario. My only concern was that I was relying on a side effect of `boolean isDirectory = fileStat.directory_p().isTrue();` to throw that error. This is probably fine, but I thought I should point it out since it could be unclear to future developers since I don't explicitly throw the error in this method.